### ISSUE TITLE
On FLOPs computation

### ISSUE THREAD

**Issue by sdeepaknarayanan** [thu-ml/stochastic_gcn#2]
Hi @cjf00000, 

First off, very interesting work. Thanks for releasing the source code.
I was looking into the appendix of your paper on Arxiv and came across the FLOPS table. I could not find the exact settings (hyperparameters, network size, number of aggregations/dense layers et cetera) for your method as well as the baselines (GraphSage). 

It would be great if you can share the exact details about the architecture.

**Comment by cjf00000** [comment]
You can find the settings in the appendix
http://proceedings.mlr.press/v80/chen18p/chen18p-supp.pdf

**Comment by sdeepaknarayanan** [comment]
Hi @cjf00000, thanks for your reply. 

I had a look at the appendix. It is still not clear if the 3 layer setting refers to 2 aggregations, followed by a dense layer or 3 aggregations followed by a dense layer. Also, the test numbers reported in Table 1 of the appendix you linked mentions 95% Testing Accuracy and 94% Validation Accuracy for all the baselines. However the text in the appendix says 94% Testing Accuracy. Can you confirm which one is correct?

**Comment by sdeepaknarayanan** [comment]
Hi @cjf00000,

Looking forward to your response.

**Comment by cjf00000** [comment]
I believe the settings in the config folder are the ones used in the paper.
https://github.com/thu-ml/stochastic_gcn/tree/master/gcn/config

I think I report the validation accuracy for all datasets, not testing accuracy. In the appendix text I reported the time to reach 94% accuracy, not 95%, because some compared algorithms can never reach 95% accuracy.

The 2-layer setting used in the main text for Reddit has exactly the same architecture as that in the GraphSAGE paper (and its codebase), except I swapped the aggregation and dropout. You can also consult to the GraphSAGE paper for the architecture. In Appendix E, I wrote for PPI and Reddit, "We use two linear layers per graph convolution layer." So there are two aggregations, and there are two dense layers per aggregation.

Except swapping the aggregation and dropout, 
For Citeseer, Cora, PubMed and NELL, I took the exact hyper-parameter and architecture from the GCN paper; for Reddit I took the exact hyper-parameter and architecture from the GraphSAGE paper. You can also reproduce my result by running my code. They are all reproducible.

**Comment by sdeepaknarayanan** [comment]
Hi @cjf00000, thanks for your response. It clarified quite a few of my doubts. 
I have one small query about FLOPs computation. Can you explain how you computed the FLOPs for Backprop operations? As far as my understanding of the code, you seem to be computing it as the same as the FLOPs for the forward pass. Can you clarify why this is the correct way?

**Comment by cjf00000** [comment]
Yeah I didn't compute the flops for the backward pass. Thanks for pointing it out, never thought of this before.

However I think computing the forward is more pessimistic than computing the whole thing, since the historical activations do not require gradient. If we compute FLOPs for the whole thing, our method would be more competitive than the batch algorithm and neighbor sampling.

**Comment by sdeepaknarayanan** [comment]
Thanks @cjf00000. I am closing the issue for now.
